ha/core/node/node_manager.py

```diff
@@ -0,0 +1,174 @@
+        Validate requested node is there in the node list
+        """
+        raise HAUnimplemented()
+
+class CortxNodeManager(NodeManager):
+    def __init__(self):
+        """
+        Prepare node instance list
+        """
+        self.node_instance_list = {}
+        self._service_manager = CortxServiceManager()
+
+        node_list = Conf.get(const.HA_GLOBAL_INDEX, "Node.nodelist")
+
+        for node_id in node_list.keys():
+            self.node_instance_list[node_id] = Node(node_id)
+
+        Log.debug(f"node_instance_list {self.node_instance_list}")
+
+        self._decision_monitor = DecisionMonitor()
+        self._refresh_context = CortxRefreshContext(self._decision_monitor)
+
+    def process_request(self, action, args, output):
+        """
+        Generic method to handle process request
+
+        Args:
+            args ([dict]): Parameter pass to request to process.
+        """
+        if action == const.NODE_COMMAND:
+            node = None
+            if args.node is None:
+                local_node = Conf.get(const.HA_GLOBAL_INDEX, "Node.local")
+                node = self.get_node_instance(local_node)
+            else:
+                node = self.get_node_instance(args.node)
+
+            self._output = output
+
+            if args.node_action == "start":
+                _action_status, _return_code = node.start()
+                self._output.output(_action_status)
+                self._output.rc(_return_code)
+
+            elif args.node_action == "stop":
+                _action_status, _return_code = node.shutdown()
+                self._output.output(_action_status)
+                self._output.rc(_return_code)
+
+            elif args.node_action == "status":
+                _action_status, _return_code = node.status()
+                self._output.output(_action_status)
+                self._output.rc(_return_code)
+
+            elif args.node_action == "refresh":
+                self._refresh_context.process_request(action, args)
+
+            else:
+                raise HAUnimplemented()
+
+        elif action == const.SERVICE_COMMAND:
+            self._service_manager.process_request(action, args, output)
+
+        else:
+            raise HAUnimplemented()
+
+    def get_node_instance_list(self):
+        """
+        Returns node list
+        """
+        return self.node_instance_list
+
+    def validate_node(self, node_id):
+        """
+        Validate requested node is there in the node list
+        """
+        valid_node = False
+
+        if node_id in self.node_instance_list.keys():
+            valid_node = True
+
+        return valid_node
+
+    def get_node_instance(self, node_id):
+        """
+        Returns instance of the node
+        """
+        if self.validate_node(node_id):
+            node = self.node_instance_list[node_id]
+        else:
+            Log.error(f"Invalid node id {node_id}")
+
+        return node
+
+class PcsNodeManager(NodeManager):
```

Comment:
class PcsNodeManager(NodeManager): 
not need in main

Reply:
Removed